src/ex5/dist/main.js

```diff
@@ -0,0 +1,242 @@
+    list.innerHTML = "";
+
+    items.forEach((item) => {
+      const listItem = document.createElement("li");
+      listItem.classList.add("list-item");
+
+      let checkBoxElement = this._createCheckBox(item);
+      checkBoxElement.classList.add("checkBox");
+
+      let divElementWithName = document.createElement("DIV");
+      divElementWithName.innerText = item.itemName;
+
+      let inputElement = document.createElement("INPUT");
+      inputElement.setAttribute("type", "text");
+      inputElement.readOnly = true;
+      inputElement.value = item.itemName;
+      inputElement.classList.add("list-item-input");
+
+      listItem.appendChild(checkBoxElement);
+      listItem.appendChild(inputElement);
+
+      if (item.isPokemon) {
+        const listItemPicture = this.getPokemonImage(item);
+        listItem.appendChild(listItemPicture);
+      }
+
+      const listItemDeleteButton = this._createDeleteButton(item);
+
+      const listItemEditButton = this._createEditButton(item);
+
+      listItem.appendChild(listItemDeleteButton);
+      listItem.appendChild(listItemEditButton);
+      listItem.setAttribute("id", item.itemId);
+      if (checkBoxElement.firstChild.checked) {
+        this.addCheckBoxStyle(listItem, item);
+      }
+      list.appendChild(listItem);
+    });
+    return "render items finished";
+  };
+  addCheckBoxStyle(listItem, item) {
+    listItem.childNodes[1].classList.add("decorate");
+    if (item.isPokemon) {
+      listItem.childNodes[2].classList.add("decorate");
+    }
+  }
+
+  getPokemonImage(pokemonData) {
+    const url = pokemonData.pokemonData;
+
+    const img = document.createElement("img");
+    img.setAttribute("src", url);
+    return img;
+  }
+
+  _createCheckBox = (item) => {
+    let divElementWithCheckBox = document.createElement("DIV");
+    let checkBox = document.createElement("INPUT");
+
+    checkBox.setAttribute("type", "checkbox");
+    checkBox.checked = item.status;
+
+    checkBox.addEventListener("change", (_) => this.handleCheckBox(item));
+
+    divElementWithCheckBox.appendChild(checkBox);
+
+    return divElementWithCheckBox;
+  };
+
+  async handleCheckBox(item) {
+    let listItem = document.getElementById(item.itemId);
+    let itemCheckBox = listItem.firstChild.firstChild;
+    let inputText = listItem.childNodes[1];
+
+    if (itemCheckBox.checked) {
+      this.addCompletedDecoration(item);
+      this.itemClient.updateStatusInDb(item.itemId, true);
+
+      let timestampNow = new Date();
+      let timestampNowHours = timestampNow.getHours();
+      timestampNow.setHours(timestampNowHours + 6);
+
+      let timestampNowToDb = timestampNow
+        .toISOString()
+        .slice(0, 19)
+        .replace("T", " ");
+    
+      timestampNow = timestampNow.toDateString();
+      inputText.value += ", " + timestampNow;
+
+      await this.itemClient.updateDoneTimestamp(item.itemId, timestampNowToDb);
+      await this.itemClient.updateNameInDb(item.itemId, inputText.value);
```

Comment:
why would we do that in 2 separate requests, what will happen if one of them fails and the other one succeeds?